### PR TITLE
fix(engine): exclude null keys from relation membership

### PR DESCRIPTION
**Pull request by carllerche**
## Summary

Negating `Profile::fields().user().in_query(User::all())` can omit profiles whose foreign key is null. Relation membership now requires present reference keys on both sides, including every component of a composite key. Apply the same exclusions to SQL and key-value backends. Simplify bound key-value row filters so empty candidate sets reduce before execution.

Add shared integration regressions for optional foreign keys, nested relations, and composite keys using the existing `in_query()` API.

## Type of change

- [x] Small / obvious change — bug fix, docs, internal cleanup, or test

## Checklist

- [x] PR title uses [Conventional Commits](https://www.conventionalcommits.org/) format
- [x] `cargo fmt` and `cargo clippy` are clean
- [x] `cargo test` passes for the engine unit tests and full SQLite and DynamoDB driver suites

## Notes for reviewers

Merge order: #1258 → #1259 → #1260 → #1261 → #1247. Each PR targets the preceding branch; rebase and retarget the next PR onto `main` after a squash merge.

This follows the relation-key rules in [the Option contract](https://github.com/tokio-rs/toasty/blob/main/docs/dev/design/option-semantics.md#edge-cases).

Validation: 393 engine unit tests, 846 SQLite tests, and 672 DynamoDB tests pass, including `filter_parent_all_ignores_unlinked_children`. Driver suites use `cargo test -p tests --test sqlite` and `cargo test -p tests --no-default-features --features dynamodb --test dynamodb -- --test-threads=1`.

### CODE REVIEW

File: crates/toasty/src/engine/lower/lift_in_subquery.rs [hunk 1]
```diff
@@ -75,19 +74,14 @@ impl<'a> LiftInSubquery<'a> {
...
 
     fn exclude_nulls(&self, expr: &mut stmt::Expr) {
-        if !self.exclude_nulls {
-            return;
-        }
-
-        let stmt::Expr::InSubquery(expr) = expr else {
+        let stmt::Expr::InSubquery(in_subquery) = expr else {
             return;
         };
```
[the remark this answers is not in the document]
Comment (carllerche) [reply crates/toasty/src/engine/lower/lift_in_subquery.rs:83]: Fixed in dcafcec01. Direct belongs-to lifts now guard nullable foreign keys, so negating membership includes rows with no related record. The new `negated_direct_belongs_to_membership` regression failed on SQLite before the fix and now passes. The full SQLite and DynamoDB suites also pass, including scoped creates.

File: crates/toasty/src/engine/lower/lift_in_subquery.rs [hunk 2]
```diff
@@ -75,19 +74,14 @@ impl<'a> LiftInSubquery<'a> {
...
-        let select = expr.query.body.as_select_mut_unwrap();
+        let select = in_subquery.query.body.as_select_mut_unwrap();
```
[the remark this answers is not in the document]
Comment (carllerche) [reply crates/toasty/src/engine/lower/lift_in_subquery.rs:90]: Fixed in dcafcec01. Limited queries select their original candidate identities before null relation keys are excluded. `limited_membership_preserves_candidates` checks membership and its negation with ordered LIMIT/OFFSET queries; it failed before the fix and now passes. Direct lifts also preserve limits and offsets that can change the result.
